ikarus/utils/observer/controllogger.cpp

```diff
@@ -9,51 +9,42 @@
 
 namespace Ikarus {
 
-void ControlLogger::updateImpl(ControlMessages message) {
-  switch (message) {
-    case ControlMessages::STEP_ENDED:
-      spdlog::info("===============================================================================");
-      break;
-    default:
-      break; // default: do nothing when notified
-  }
-}
-
-void ControlLogger::updateImpl(ControlMessages message, const std::string& pathFollowingName) {
+void ControlLogger::updateImpl(ControlMessages message, const ControlState& state) {
   switch (message) {
     case ControlMessages::CONTROL_STARTED:
       start_ = std::chrono::high_resolution_clock::now();
-      spdlog::info("===============================================================================");
-      spdlog::info("Started path following with: {}", pathFollowingName);
-      spdlog::info("===============================================================================");
-      break;
-    default:
-      break;
-  }
-}
-
-void ControlLogger::updateImpl(ControlMessages message, int stepNumber, double stepSize) {
-  switch (message) {
-    case ControlMessages::STEP_STARTED:
-      spdlog::info("Load step: {:>4} {:>49} {:<.2e}", stepNumber, "Step size = ", stepSize);
-      spdlog::info("-------------------------------------------------------------------------------");
-      break;
-    default:
+      spdlog::info("=====================================================================");
+      spdlog::info("Started {}", state.name);
+      spdlog::info("=====================================================================");
       break;
-  }
-}
-
-void ControlLogger::updateImpl(ControlMessages message, int totalIterations, const std::string& pathFollowingName) {
-  switch (message) {
     case ControlMessages::CONTROL_ENDED:
       stop_     = std::chrono::high_resolution_clock::now();
       duration_ = duration_cast<std::chrono::milliseconds>(stop_ - start_);
-      spdlog::info("End of path following with {} control", pathFollowingName);
-      spdlog::info("Total number of iterations: {:3d}", totalIterations);
+      spdlog::info("End of {}", state.name);
+      spdlog::info("Total number of iterations: {:3d}", state.totalIterations);
       spdlog::info("Elapsed time: {} ms", duration_.count());
       break;
-    default:
+    case ControlMessages::STEP_STARTED:
+      if (state.initialConfig) {
```

Comment:
A boolean to flag if the configuration is an initial configuration is introduced such that here the output is differentiated and is reasonable, else the numbers displayed on the screen can be misleading